tests/epyccel/test_epyccel_mod.py

```diff
@@ -11,38 +11,68 @@ def modulo_i_i(x, y):
         return x % y
 
     f = epyccel(modulo_i_i, language=language)
-    x = randint(-1e6, 1e6)
+    x = randint(0, 1e6)
+    nx = randint(-1e6, -1)
     y = randint(low=-100, high=100)
     y = 1 if y == 0 else y
 
-    assert f(x, y) == modulo_i_i(x, y)
-    assert isinstance(f(x, y), type(modulo_i_i(x, y)))
+
+    f_output = f(x, y)
+    modulo_i_i_output = modulo_i_i(x, y)
+    assert modulo_i_i_output == f_output
+    assert isinstance(f_output, type(modulo_i_i_output))
+
+    #test negative x
+    f_output = f(nx, y)
+    modulo_i_i_output = modulo_i_i(nx, y)
+    assert modulo_i_i_output == f_output
+    assert isinstance(f_output, type(modulo_i_i_output))
 
 def test_modulo_real_real(language):
     @types('real', 'real')
     def modulo_r_r(x, y):
         return x % y
 
     f = epyccel(modulo_r_r, language=language)
-    x = uniform(low=-1e6, high=1e6)
+    x = uniform(low=0 high=1e6)
+    nx = uniform(low=-1e6, high=-1)
     y = uniform(low=-1e2, high=1e2)
     y = 1 if y == 0 else y
 
-    assert(isclose(f(x, y), modulo_r_r(x, y), rtol=1e-15, atol=1e-15))
-    assert isinstance(f(x, y), type(modulo_r_r(x, y)))
+    f_output = f(x, y)
+    modulo_r_r_output = modulo_r_r(x, y)
+    assert(isclose(f_output, modulo_r_r_output, rtol=1e-15, atol=1e-15))
+    assert isinstance(f_output, type(modulo_r_r_output))
+
+    # test negative x
+    f_output = f(nx, y)
+    modulo_r_r_output = modulo_r_r(nx, y)
+    assert(isclose(f_output, modulo_r_r_output, rtol=1e-15, atol=1e-15))
+    assert isinstance(f_output, type(modulo_r_r_output))
 
 def test_modulo_real_int(language):
     @types('real', 'int')
     def modulo_r_i(x, y):
         return x % y
 
     f = epyccel(modulo_r_i, language=language)
-    x = uniform(low=-1e6, high=1e6)
+    x = uniform(low=0, high=1e6)
+    nx = uniform(low=-1e6, high=-1)
     y = randint(low=-100, high=100)
     y = 1 if y == 0 else y
 
-    assert(isclose(f(x, y), modulo_r_i(x, y), rtol=1e-15, atol=1e-15))
-    assert isinstance(f(x, y), type(modulo_r_i(x, y)))
+
+    f_output = f(x, y)
+    modulo_r_i_output = modulo_r_i(x, y)
+    assert(isclose(f_output, modulo_r_i_output, rtol=1e-15, atol=1e-15))
```

Comment:
This doesn't seem to have fixed the problem as we can see on linux. The error message is:
```
>       assert(isclose(f(x, y, z), modulo_multiple(x, y, z), rtol=RTOL, atol=ATOL))
E       assert False
E        +  where False = isclose(8.964345582700844, 8.964345582702663, rtol=1e-14, atol=1e-15)
E        +    where 8.964345582700844 = <built-in function modulo_multiple>(309160, 9399.40736420054, 93)
E        +    and   8.964345582702663 = <function test_modulo_multiple.<locals>.modulo_multiple at 0x7fcc64c07c80>(309160, 9399.40736420054, 93
```
A quick test shows that for these values to be considered "close" the tolerances would need to be:
- rtol=1e-12
- atol=1e-13

This is quite a high tolerance, especially for a basic operator so I don't really want to use this tolerance if it can be avoided.

Maybe we could try an IfTernaryOperator?
E.g. `x % y` could be translated to `x < 0 ? x % y + y : x % y`

But I am not sure to what extent this would slow down the code or impact it's readability.

I already think that the translation of:
```
return x % y % z
```
to:
```
return (fmod(((fmod((fmod(((fmod((double)(x), y)) + y), y)), (double)(z))) + (double)(z)), (double)(z)));
```
gives code which is much harder to understand. For 1 modulo operator it seems ok, but as soon as there is more happening in the line it quickly becomes hard to read.

Imagine if we make things worse by making `y` an indexed element, e.g.:
```
return x % y[0] % z
```
now we generate
```
return (fmod(((fmod((fmod(((fmod((double)(x), y.nd_double[get_index(y, 0)])) + y.nd_double[get_index(y, 0)]), y.nd_double[get_index(y, 0)])), (double)(z))) + (double)(z)), (double)(z)));
```
personally I cannot understand this line.
I suspect we ought to make the modulo operator a function in our internal library in order to avoid this kind of obfuscation. Maybe we can discuss this in the scrum today. I would like to get a second opinion

Reply:
> Maybe we could try an `IfTernaryOperator`?
> E.g. `x % y` could be translated to `x < 0 ? x % y + y : x % y`
> 
> But I am not sure to what extent this would slow down the code or impact it's readability.

As per [here](https://stackoverflow.com/a/28889216/5957195), this should not slow down the code too much, however the condition for the `IfTernaryOperator` is more complicated than above:
`((x < 0 and y > 0) or (x > 0 and y < 0)) ? x % y + y : x % y`

Reply:
Correction:
XOR can be more succinctly written as:
`((x > 0) != (y > 0)) ? x % y + y : x % y`